app/assets/javascripts/models/ui/state_manager.coffee

```diff
@@ -142,15 +143,23 @@
 
       root = '/search'
 
-      return root if state.parent && state.current != 'collection-details'
+      if state.parent && state.current != 'collection-details'
+        if state.current == 'granule-list' || state.current == 'project-overview' || state.current == 'granule-details'
```

Comment:
I think I'd prefer to see all of these if-statements (lines 147, 155, 160) combined into a single if-statement on line 144 that says something like:

    state.parent = false unless state.current == 'collection-results'